draft-ietf-quic-multipath.md

```diff
@@ -1353,6 +1338,23 @@ Maximum Path Identifier:
   Receipt of a value that is higher than the local maximum value MUST
   be treated as a connection error of type PROTOCOL_VIOLATION.
 
+# Error Codes {#error-codes}
+
+Multipath QUIC transport error codes are 62-bit unsigned integers
+following {{QUIC-TRANSPORT}}.
+
+The following additional QUIC error codes are defined for use in the PATH_ABANDON frame.
```

Comment:
```suggestion
In addition to NO_ERROR(0x0) following {{QUIC-TRANSPORT}}, the following QUIC error codes are defined for use in the PATH_ABANDON frame.
```

Reply:
While I agree that NO_ERROR is a fine error message for a PATH_ABANDON, the current text is not really clear about the other error codes defined in RFC9000. For instance, what should a receiver do if it receives a PATH_ABANDON with error code FLOW_CONTROL_ERROR? This would sound wrong to me, though it is not clear how a receiver would react.

Reply:
I don't think the receiver would react it all. This is only for logging. If there are concerns about this that I think we would need to define a separate code space. Thinking more about this, I think using the QUIC error code space is wrong. I think we should not call this error code at all. In case of connection close, you actually close the connection because there was an unexpected protocol behavior. In case of the multipath extension, it is the same: if there is something unexpected, you have to close the whole connection. Therefore closing a path is never an error but always an explicit decision.

Reply:
Yes, I agree with your point. If we need such logging, we may standardise a few PATH_ABANDON-specific reason code points (the three mentioned above seem sufficient).

Reply:
As a reference, [HTTP/3's error code space](https://www.iana.org/assignments/http3-parameters/http3-parameters.xhtml#http3-parameters-error-codes) contains codes that cover various scenarios.

This include unexpected protocol errors, such as detecting the peer closed a stream it shouldn't and responding with a CONNECTION_CLOSE of H3_CLOSED_CRITICAL_STREAM.

But it also includes expected behaviours that are not errors, such as when an endpoint cancels a request (a bit like path abandon) by sending a REST_STREAM or STOP_SENDING using H3_REQUEST_CANCELLED.

In all these cases, an H3 implementation has space to ignore the specific language in the spec: use the wrong error code space (send a transport error when it was an application error), or send a generic code, or send an incorrect code. I don't think any of those cause any real problems for the recipient, as Mirja put it, it's mainly for debugging. And someone investigating these should always be skeptical of the peer not telling the ground truth.

On balance, I think its fine to use the QUIC error code space here. This is an example of an extension where we are building some more complexity into the transport and can expect to need to extend the shared codepoint space to suit.

Reply:
I also think that using the QUIC error space is fine. It is fairly obvious that not all error codes defined in the QUIC error space apply to all scenarios. For example, a "transport parameter error" only makes sense when parsing transport parameters. The important property is that the error codes be unique.

I also do not think that any implementation should react to specific error code values, unless that's explicitly mentioned in the text of the standard. The purpose is documentation for helping debugging issues, or for passing information to the application and ultimately the user. We don't need a special registry for path abandon reasons.